src/components/ImageWithFallback.tsx

```diff
@@ -1,12 +1,17 @@
 import { useState } from 'react'
 import Image, { ImageProps } from 'next/image'
+import { useInView } from 'react-intersection-observer'
 
 export const ImageWithFallback = ({ fallbackSrc, ...props }: { fallbackSrc: string } & ImageProps) => {
   const [imageSrc, setImageSrc] = useState(props.src)
+  const { ref, inView } = useInView({
+    triggerOnce: true,
+    rootMargin: '200px 0px',
+  })
 
   const onError = () => {
     setImageSrc(fallbackSrc)
   }
 
-  return <Image {...props} src={imageSrc} alt={props.alt} onError={onError} />
+  return <div ref={ref}>{inView ? <Image {...props} src={imageSrc} alt={props.alt} onError={onError} /> : null}</div>
```

Comment:
`<Image/>` from nextjs supports lazy loading. Why do we need this?
https://nextjs.org/docs/pages/api-reference/components/image#loading
Plus, switching between null (zero-size) and Image (fixed-size) will cause flickering